main/src/ca/uwaterloo/flix/language/phase/Kinder.scala

```diff
@@ -72,39 +72,29 @@ object Kinder {
     flatMapN(visitTypeAliases(root.taOrder, root)) {
       taenv =>
 
-        // Extra type annotations are required due to limitations in Scala's type inference.
-        val enumsVal = Validation.sequence(ParOps.parMap(root.enums)({
-          pair: (Symbol.EnumSym, ResolvedAst.Declaration.Enum) =>
-            val (sym, enum) = pair
-            visitEnum(enum, taenv, root).map(sym -> _)
-        }))
-
-        // Extra type annotations are required due to limitations in Scala's type inference.
-        val restrictableEnumsVal = Validation.sequence(ParOps.parMap(root.restrictableEnums)({
-          pair: (Symbol.RestrictableEnumSym, ResolvedAst.Declaration.RestrictableEnum) =>
-            val (sym, enum) = pair
-            visitRestrictableEnum(enum, taenv, root).map(sym -> _)
-        }))
+        val enumsVal = ParOps.mapValuesFallible(root.enums) {
```

Comment:
this  can become `....mapValues(root.enums)(visitEnum(_, taenv, root)`.

Same pattern can be used everywhere else.